### ISSUE TITLE
inference code

### ISSUE THREAD

**Issue by lkcl0928** [hongyliu/CLRNet#3]
import torch
import cv2
from clrnet.engine.runner import Runner
from clrnet.utils.config import Config
import cv2
import os
import time
import torch.backends.cudnn as cudnn
cfg = Config.fromfile("configs/clrnet/clr_resnet18_culane.py")
cfg.gpus = 1
cfg.load_from = 'culane_r18.pth'
cfg.resume_from = ''
cfg.finetune_from = ''
cfg.seed = 0
def pre_process_image(path):
    img_w = 800
    img_h = 320
    cut_height = 270 
    image = cv2.imread(path)
    image = image[cut_height:, :, :]
    image = cv2.resize(image, (img_w, img_h))
    image = torch.Tensor([image])
    image = image.permute(0, 3, 1, 2)
    return image
def view(predictions):
        for lanes in predictions:
            img = cv2.imread("/root/CLRNet/v2-b40621ccca1d9a432adf1442dcc69540_b.jpg")
            out_file = "1.jpg"
            lanes = [lane.to_array(cfg) for lane in lanes]
            imshow_lanes(img, lanes, out_file=out_file)
def imshow_lanes(img, lanes, show=False, out_file=None, width=4):
    lanes_xys = []
    for _, lane in enumerate(lanes):
        xys = []
        for x, y in lane:
            if x <= 0 or y <= 0:
                continue
            x, y = int(x), int(y)
            xys.append((x, y))
        lanes_xys.append(xys)
    lanes_xys.sort(key=lambda xys : xys[0][0])

    for idx, xys in enumerate(lanes_xys):
        for i in range(1, len(xys)):
            cv2.line(img, xys[i - 1], xys[i], COLORS[idx], thickness=width)
    if show:
        cv2.imshow('view', img)
        cv2.waitKey(0)
    if out_file:
        cv2.imwrite(out_file, img)
t0=time.time()
runner = Runner(cfg)
runner.net.eval()
predictions = []
image = pre_process_image("/root/CLRNet/v2-b40621ccca1d9a432adf1442dcc69540_b.jpg")
data = image.cuda()
with torch.no_grad():
    output = runner.net(data)
    output = runner.net.module.heads.get_lanes(output)
    print(output)
    t1=time.time()
    predictions.extend(output)
    view(output)
    print((t1-t0))
Hi, I write the inference code , but it does not show any results.output  returns a empty list
Appreciate any help?

**Comment by lmadkm** [comment]
Hello, I have the same question but with the benchmarks (CULane&TuSimple) testing, where the output = self.net(data) is so small in about 10^*-3~10^*-4 level, resulting in the output = self.net.heads.get_lanes(output) is the empty one. How did you deal with your problem? Waiting for your reponses~
